frontend/src/scenes/experiments/ExperimentView/CumulativeExposuresChart.tsx

```diff
@@ -0,0 +1,98 @@
+import { useValues } from 'kea'
+import { InsightEmptyState } from 'scenes/insights/EmptyStates'
+
+import { InsightViz } from '~/queries/nodes/InsightViz/InsightViz'
+import { queryFromFilters } from '~/queries/nodes/InsightViz/utils'
+import { InsightQueryNode, InsightVizNode, NodeKind } from '~/queries/schema'
+import { BaseMathType, ChartDisplayType, Experiment, InsightType, PropertyFilterType, PropertyOperator } from '~/types'
+
+import { experimentLogic } from '../experimentLogic'
+
+const getCumulativeExposuresQuery = (experiment: Experiment): InsightVizNode<InsightQueryNode> => {
+    const experimentInsightType = experiment.filters?.insight || InsightType.TRENDS
+
+    const variants = experiment.parameters?.feature_flag_variants?.map((variant) => variant.key) || []
+    if (experiment.holdout) {
+        variants.push(`holdout-${experiment.holdout.id}`)
+    }
+
+    // Trends Experiment
+    if (experimentInsightType === InsightType.TRENDS && experiment.parameters?.custom_exposure_filter) {
+        const queryFilters = {
+            ...experiment.parameters?.custom_exposure_filter,
+            display: ChartDisplayType.ActionsLineGraphCumulative,
+        }
+        return queryFromFilters(queryFilters)
+    }
```

Comment:
The raw exposure query won't work out of the box. We need to add to it:
- the properties to only include the events with the valid variants
- the breakdown
- the experiment start and end date

This is exactly what we're doing in the backend to get the final exposure query. We need to replicate it here too: https://github.com/PostHog/posthog/blob/master/ee/clickhouse/queries/experiments/trend_experiment_result.py#L162-L185

The following modified query works correctly:
```
        const queryFilters = {
            ...experiment.parameters?.custom_exposure_filter,
            date_from: experiment.start_date,
            date_to: experiment.end_date,
            breakdown: `$feature/${experiment.feature_flag_key}`,
            properties: [
                {
                    key: `$feature/${experiment.feature_flag_key}`,
                    value: variants,
                    operator: 'exact',
                    type: 'event',
                },
            ],
            display: ChartDisplayType.ActionsLineGraphCumulative,
        } as FilterType
```

![image](https://github.com/user-attachments/assets/968d5f5f-541d-4291-8025-b9571dd23afb)

---

Actually, thinking about it more, there's a better way to do this! In the backend, we enrich the raw exposure query. This is exactly the query we then use for our results calculation. Why don't we just return this enriched query to the frontend and simply pass it to this component? This will be much less prone to errors.

We actually do something similar already for the main events query - [we return `filters` in the experiment results](https://github.com/PostHog/posthog/blob/master/ee/clickhouse/queries/experiments/trend_experiment_result.py#L236) - these are the enriched filters used in the results calculation, and we then [pass these filters to the `<Query />` component](https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/experiments/ExperimentView/components.tsx#L173-L197) for the result visualization.

We can simply do the same for the exposure filters, and pass them here.

---

Note that the above will only apply to the Trends experiments. For Funnels experiments, there's no exposure query, so we'll need to construct that query here.

Reply:
> Actually, thinking about it more, there's a better way to do this! In the backend, we enrich the raw exposure query. This is exactly the query we then use for our results calculation. Why don't we just return this enriched query to the frontend and simply pass it to this component? This will be much less prone to errors.

4de4047c5592a8c6f2036ab7d6a30ba9a2ee552a

Reply:
flyby without looking deeper, we also calculate the exposure on the backend, why not send the results back instead of the filters? shouldn't need to do the same work again, given its the same query.

Reply:
Hmm could do. I see that in the main results we're passing both filters and the results. I'll do this in a follow up after we migrate to HogQL.

```
        <Query
            query={{
                kind: NodeKind.InsightVizNode,
                source: filtersToQueryNode(transformResultFilters(oldQueryResults?.filters ?? {})),
                showTable,
                showLastComputation: true,
                showLastComputationRefresh: false,
            }}
            context={{
                insightProps: {
                    dashboardItemId: oldQueryResults?.fakeInsightId as InsightShortId,
                    cachedInsight: {
                        short_id: oldQueryResults?.fakeInsightId as InsightShortId,
                        query: oldQueryResults?.filters
                            ? queryFromFilters(transformResultFilters(oldQueryResults.filters))
                            : null,
                        result: oldQueryResults?.insight,
                        disable_baseline: true,
                        last_refresh: oldQueryResults?.last_refresh,
                    },
                    doNotLoad: true,
                },
            }}
            readOnly
        />
```